src/h3lib/lib/h3UniEdge.c

```diff
@@ -229,14 +242,25 @@ void H3_EXPORT(getH3UnidirectionalEdgeBoundary)(H3Index edge, GeoBoundary* gb) {
 
     int k = 0;
     for (int i = 0; i < origin.numVerts; i++) {
-        for (int j = 0; j < destination.numVerts; j++) {
-            if (geoAlmostEqualThreshold(&origin.verts[i], &destination.verts[j],
-                                        0.000001)) {
-                gb->verts[k].lat = origin.verts[i].lat;
-                gb->verts[k].lon = origin.verts[i].lon;
+        if (_hasMatchingVertex(&origin.verts[i], &destination)) {
+            // If we are on vertex 0, we need to handle the case where it's the
+            // end of the edge, not the beginning.
+            // TODO: This is fine for this limited case, but it might be
+            // better/cleaner to use a VertexGraph for this
```

Comment:
You are better off avoiding allocations involved in creating a `VertexGraph` and hacking a solution with only the given inputs.

Reply:
Yeah, probably true. It just irks me slightly that this is basically the same thing we're doing in `h3SetToVertexGraph` but we're doing it totally differently :/. Also bothers me that this is `O(n^2)`, even though `n` is guaranteed to be <= 10 and is usually 6. I'll drop the comment though, I doubt there'd be any benefit.